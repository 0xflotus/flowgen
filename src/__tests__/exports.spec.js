// @flow

import { compiler, beautify } from "..";

it("should handle exports", () => {
  const ts = `
export default module
export { module }
export { GeneratorOptions } from "@babel/generator";
export * from 'typescript';
//enable when typescript supports
//export traverse, { Visitor, NodePath } from "@babel/traverse";
//export template from "@babel/template";
//export * as t from "@babel/types";`;
  const result = compiler.compileDefinitionString(ts);
  expect(beautify(result)).toMatchSnapshot();
});
