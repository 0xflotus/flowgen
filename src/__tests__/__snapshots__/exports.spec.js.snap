// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should handle exports 1`] = `
"declare export default typeof module;
declare export { module };
declare export { module as newModule };
declare export {
  GeneratorOptions as NewGeneratorOptions
} from \\"@babel/generator\\";
declare export * from \\"typescript\\"
"
`;
